File: scripts/ws_probe.py
import asyncio
import json
import sys

import websockets


async def main(ticket_id: str, message: str) -> int:
    uri = f"ws://localhost:8000/ws/{ticket_id}"
    async with websockets.connect(uri, open_timeout=10, close_timeout=5) as ws:
        await ws.send(json.dumps({"message": message, "user": "Codex"}))
        for _ in range(30):
            try:
                payload = await asyncio.wait_for(ws.recv(), timeout=5)
            except asyncio.TimeoutError:
                print("TIMEOUT_WAITING_FOR_MESSAGE")
                return 2
            print(payload)
            lowered = payload.lower()
            if "stream_end" in lowered or "run_finished" in lowered or '"type":"error"' in lowered:
                return 0
    return 0


if __name__ == "__main__":
    raise SystemExit(asyncio.run(main(sys.argv[1], sys.argv[2])))

File: scripts/ws-probe.mjs
const ticketId = process.argv[2];
const message = process.argv[3];

if (!ticketId || !message) {
  console.error("usage: node scripts/ws-probe.mjs <ticketId> <message>");
  process.exit(1);
}

const ws = new WebSocket(`ws://localhost:8000/ws/${ticketId}`);

const timeout = setTimeout(() => {
  console.log("TIMEOUT_WAITING_FOR_MESSAGE");
  try { ws.close(); } catch {}
  process.exit(2);
}, 30000);

ws.addEventListener("open", () => {
  ws.send(JSON.stringify({ message, user: "Codex" }));
});

ws.addEventListener("message", (event) => {
  const payload = String(event.data);
  console.log(payload);
  const lowered = payload.toLowerCase();
  if (
    lowered.includes("stream_end") ||
    lowered.includes("run_finished") ||
    lowered.includes('"type":"error"')
  ) {
    clearTimeout(timeout);
    try { ws.close(); } catch {}
    process.exit(0);
  }
});

ws.addEventListener("error", (event) => {
  clearTimeout(timeout);
  console.error("WS_ERROR", event?.message || "");
  process.exit(3);
});
